samples_test/build.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use std::{
+    env,
+    fs::{read_dir, File},
+    io::Write,
+    path::Path,
+};
+
+fn main() {
+    println!("cargo::rerun-if-changed=../samples/algorithms/");
+    // Iterate through the samples folder and create a test for each file
+    let mut paths =
+        read_dir("../samples/algorithms/").expect("folder should exist and be readable");
+    let out_dir = env::var("OUT_DIR").expect("OUT_DIR should be set");
+    let dest_path = Path::new(&out_dir).join("test_cases.rs");
+    let mut f = File::create(dest_path).expect("files should be creatable in OUT_DIR");
+
+    while let Some(Ok(dir_entry)) = paths.next() {
+        let path = &dir_entry.path();
+        let file_name = path
+            .file_name()
+            .expect("file name should be separable")
+            .to_str()
+            .expect("file name should be valid");
+        let file_stem = path
+            .file_stem()
+            .expect("file name should be separable")
+            .to_str()
+            .expect("file name should be valid");
+        let file_stem_upper = file_stem.to_uppercase();
```

Comment:
Can we check for spaces in the file name? Sounds dumb, but the first thing I tried is to copy-paste `BellState.qs` and ended up with `BellState copy.qs` in my samples folder. Might be confusing when you're just playing around with samples, unaware that it's part of the build process.

![image](https://github.com/microsoft/qsharp/assets/16928427/754ef821-5ec5-4492-9b77-c5f4ea4a3eb1)

Reply:
Added an assert, so now it fails early with the offending file name:
![image](https://github.com/microsoft/qsharp/assets/10567287/2b63a7b5-ed21-4c6a-a290-60b55a2fd4d2)